src/compiler/value/arithmetic.rs

```diff
@@ -107,18 +112,20 @@ impl VrlValueArithmetic for Value {
     fn try_add(self, rhs: Self) -> Result<Self, ValueError> {
         let value = match (self, rhs) {
             (Value::Integer(lhs), Value::Float(rhs)) => Value::from_f64_or_zero(lhs as f64 + *rhs),
-            (Value::Integer(lhs), rhs) => (lhs
-                + rhs
+            (Value::Integer(lhs), rhs) => {
+                let rhv = rhs
                     .try_into_i64()
-                    .map_err(|_| ValueError::Add(Kind::integer(), rhs.kind()))?)
-            .into(),
+                    .map_err(|_| ValueError::Add(Kind::integer(), rhs.kind()))?;
+                i64::wrapping_add(lhs, rhv).into()
+            }
             (Value::Float(lhs), rhs) => (lhs
                 + rhs
                     .try_into_f64()
                     .map_err(|_| ValueError::Add(Kind::float(), rhs.kind()))?)
             .into(),
             (lhs @ Value::Bytes(_), Value::Null) => lhs,
             (Value::Bytes(lhs), Value::Bytes(rhs)) => {
+                #[allow(clippy::integer_arithmetic)]
                 let mut value = BytesMut::with_capacity(lhs.len() + rhs.len());
```

Comment:
I wonder if we wouldn't actually _want_ this to panic. The behavior otherwise seems like it'd be strange.

Reply:
Today you could never get into a situation where this would overflow. If either of these values are large enough to cause an overflow, that means you would have already allocated several million terrabytes :)


In general though, I'd like to add some kind of allocation limit in VRL for https://github.com/vectordotdev/vrl/issues/242